sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVInferSchema.scala

```diff
@@ -78,7 +79,63 @@ private[csv] object CSVInferSchema {
    * Infer type of string field. Given known type Double, and a string "1", there is no
    * point checking if it is an Int, as the final type must be Double or higher.
    */
-  def inferField(typeSoFar: DataType, field: String, nullValue: String = ""): DataType = {
+  def inferField(typeSoFar: DataType,
+      field: String,
+      nullValue: String = "",
+      dateFormat: SimpleDateFormat = null): DataType = {
+    def tryParseInteger(field: String): DataType = if ((allCatch opt field.toInt).isDefined) {
```

Comment:
I see.. Actually, I did not move them for no reason. It was becuase `tryParseTimestamp()` refers `dateFormat` but did not want to add another argument.
Anyway, let me make this like the original but taking another argument (`CSVOptions`).